lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -543,7 +543,7 @@ def get_mount_facts(self):
 
             device, mount, fstype, options = fields[0], fields[1], fields[2], fields[3]
 
-            if not device.startswith(('/', '\\')) and ':/' not in device or fstype == 'none':
+            if not device.startswith(('/', '\\')) and ':/' not in device and fstype != 'zfs' or fstype == 'none':
```

Comment:
This conditional is getting too brittle and complicated. This test needs to be moved into a function.

Reply:
The idea behind this originally was to skip network mounts because they cause fact gathering to hang. I don't believe ZFS mounts would have this problem because they are local, correct? I just want to ensure the original intention of this code is preserved as we start enumerating more mounts. Maybe a comment should be added explaining this.

Reply:
> I'm not very familiar with `zfs`. Could you provide more examples of `zfs` mounts for the unit test, or do they all pretty much follow that same format?

ZFS mounts have the following format (in Linux at least):

`[a-z]([a-z0-9.:_-]*/)+ <mountpoint> zfs <options> 0 0`

I can add more examples, using some of the special characters (`.`, `:`).

Reply:
> The idea behind this originally was to skip network mounts because they cause fact gathering to hang. I don't believe ZFS mounts would have this problem because they are local, correct? I just want to ensure the original intention of this code is preserved as we start enumerating more mounts. Maybe a comment should be added explaining this.

Does the current condition skip only network mounts or other types as well (the `fstype=none` part)?

Also, the current condition does not skip the following mount, which is a network one:

`grimlock.g.a:/mnt/data/foto's /home/adrian/fotos fuse.sshfs rw,nosuid,nodev,relatime,user_id=1000,group_id=1000 0 0`

Actually, what is the reasoning behind the `':/' not in device` part of the condition? Because ':/' is commonly found in network mount device names.

Reply:
I've been looking at this and I'm not entirely sure. I went through the history of that conditional and didn't find any commits explaining it.

My current understanding is the conditional is meant to avoid running `self.get_mount_info()` on mounts that would cause fact gathering to hang. This is mostly network mounts but could also include mounts that have a filesystem type of `none`. I think it should fine to proceed making a checker function with this goal in mind.

> Also, the current condition does not skip the following network mount from test data:

I agree that would not be skipped by the current conditional. However, it looks like that line is only used by the `test_get_mtab_entries` test, not the `test_get_mount_facts` test. That seems like a test case should be added to the `MTAB_ENTRIES` fixture.

I'll be out next week but will follow up when I get back. Thanks for working on this.

Reply:
> My current understanding is the conditional is meant to avoid running `self.get_mount_info()` on mounts that would cause fact gathering to hang. This is mostly network mounts but could also include mounts that have a filesystem type of `none`. I think it should fine to proceed making a checker function with this goal in mind.

I have refactored the condition into a function, giving it some context in the comment. As a part of that, I have removed the `':/'` part of the condition entirely, because it's commonly found within network mounts and because I could not find an example of a local filesystem on Linux (other than ZFS) where the device name would not start with a slash and included `':/'`. So, I believe `fstype != 'zfs'` is sufficient.

> I agree that would not be skipped by the current conditional. However, it looks like that line is only used by the `test_get_mtab_entries` test, not the `test_get_mount_facts` test. That seems like a test case should be added to the `MTAB_ENTRIES` fixture.

I have expanded `test_get_mounts_facts`  to check whether the parsed mounts are the same as what we expect, given `MTAB_ENTRIES`. I have listed the valid mounts in a separate constant, so `MTAB_ENTRIES` syntax remains the same as `_mtab_entries` output.